tsdb/head_append.go

```diff
@@ -357,20 +357,22 @@ func (a *headAppender) Append(ref storage.SeriesRef, lset labels.Labels, t int64
 	}
 
 	if value.IsStaleNaN(v) {
-		// This is not thread safe as we should be holding the lock for "s".
+		s.Lock()
```

Comment:
nit: for simplicity and assuming it's ok to run `value.IsStaleNaN(v)` with the code held, we can move this before the `if` (if not, maybe add a comment so it's not moved for that reason in the future)

Reply:
It looks indeed that we can move the `s.Lock` outside of the `if` (and remove the `else` branch below).

Maybe I'm missing the same thing as @machine424 , but then it definitely warrants a comment.

Reply:
Agreed; I posted the simplification.  Please squash if merging.